lib/plugin-installer.js

```diff
@@ -0,0 +1,274 @@
+const fs = require('fs');
+const os = require('os');
+const path = require('path');
+const { spawn } = require('child_process');
+const YAWN = require('yawn-yaml/cjs');
+
+class PluginInstaller {
+  constructor () {
+    this.pluginManifest = null;
+    this.packageName = null;
+  }
+
+  static get PACKAGE_PREFIX () {
+    return 'express-gateway-plugin-';
+  }
+
+  static create () {
+    return new PluginInstaller();
+  }
+
+  install (options) {
+    return this.runNPMInstallation()
+      .then(() => this.updateConfigurationFiles());
+  }
+
+  runNPMInstallation ({ packageSpecifier, cwd, env }) {
+    return new Promise((resolve, reject) => {
+      // manually spawn npm
+      // use --parseable flag to get tab-delimited output
+      // forward sterr to process.stderr
+      // capture stdout to get package name
+
+      let pluginPath = null;
+
+      const installArgs = [
+        'install', packageSpecifier,
+        '--cache-min', 24 * 60 * 60,
+        '--parseable'
+      ];
+
+      const installOpts = {
+        cwd: cwd || process.cwd(),
+        env: env || process.env,
+        stdio: ['ignore', 'pipe', 'inherit']
+      };
+
+      const npmInstall = spawn('npm', installArgs, installOpts);
+
+      npmInstall.on('error', _ => {
+        reject(new Error('Cannot install', packageSpecifier));
+      });
+
+      const bufs = [];
+      let len = 0;
+      npmInstall.stdout.on('readable', () => {
+        const buf = npmInstall.stdout.read();
+
+        if (buf) {
+          bufs.push(buf);
+          len += buf.length;
+        }
+      });
+
+      npmInstall.stdout.on('end', () => {
+        const lines = Buffer.concat(bufs, len)
+          .toString()
+          .trim()
+          .split('\n');
+
+        const output = lines[lines.length - 1].split('\t');
+
+        if (output.length < 4) {
+          reject(new Error('Cannot parse npm output while installing plugin.'));
+          process.exit();
+        }
+
+        this.packageName = output[1];
+        pluginPath = path.join(cwd, output[3]);
+      });
+
+      npmInstall.on('exit', () => {
+        if (pluginPath) {
+          this.pluginManifest = require(pluginPath);
+
+          resolve({
+            packageName: this.packageName,
+            pluginManifest: this.pluginManifest
+          });
+        }
+      });
+    });
+  }
+
+  get existingPluginOptions () {
+    const config = require('./config');
+    const systemConfig = config.systemConfig;
+
+    const name = this.pluginKey;
+
+    const existingPluginOptions =
+      systemConfig.plugins && systemConfig.plugins[name]
+        ? systemConfig.plugins[name] : {};
+
+    return existingPluginOptions;
+  }
+
+  get pluginKey () {
+    let name = this.pluginManifest.name || this.packageName;
+    let abbreviatedPackageName = null;
+
+    if (this.packageName.startsWith(PluginInstaller.PACKAGE_PREFIX)) {
```

Comment:
I was trying to install `ExpressGateway/express-gateway-plugin-example#master` that should work 
some flag needed to avoid prefixing

Reply:
I think this bit of code is doing its job just fine.  There's no code path to add a non-aliased plugin to the config.  That was because YAWN can't handle the empty object.  We can either add `package: <packagename>` or `disabled: false` to make it work... at least until we fix YAWN.

Reply:
add `package: name` only when there is package
and don't add this property at all if name is the same as package, or I'm missing something ?

Reply:
```yaml
example:

```

That won't parse in YAWN.

```yaml
example:
  disabled: false
```

```yaml
example:
  package: express-gateway-plugin-example
```

Those will parse in YAWN.

Reply:
_sigh_ 
let's use `package:` for now to be explicit even when installing by convention and we can deal with the ramifications later

Reply:
Addressed in the latest commit.